src/include/duckdb/execution/aggregate_hashtable.hpp

```diff
@@ -173,6 +174,9 @@ class GroupedAggregateHashTable : public BaseAggregateHashTable {
 
 	vector<ExpressionType> predicates;
 
+	//! The arena allocator used by the aggregates for their internal state
+	ArenaAllocator aggregate_allocator;
```

Comment:
What if we have many distinct groups and have to use partitioning? Won't the `ArenaAllocator` of the initial (unpartitioned) HT still be the owner of the allocated data?

Although I assume the receiving aggregate state is responsible for ownership of the data from the giving aggregate state when they are combined by, e.g., copying it. For the `list` aggregate function we use a linked list, so in theory, we could do a zero-copy combine.

Reply:
The `Combine` currently always copies over the data and keeps the original data in-tact, which allows this to work as-is. For a zero-copy combine we would need to move over the allocators as well, but I will leave that for a future PR.

Reply:
You are only allowed to take ownership of the incoming state during `combine` if you know you are not being called by the windowing logic (the segment tree code uses combine on elements of the tree). One notable (and relevant) example is the sorted aggregate wrapper, which is only used by regular aggregation. This will probably not change if we add `ORDER BY` argument clauses to windowed aggregates as I expect we will use Adrian's stuff instead.